src/main/java/com/solopov/urlshorter/web/dto/UrlGenerateRequestDto.java

```diff
@@ -0,0 +1,16 @@
+package com.solopov.urlshorter.web.dto;
+
+import com.solopov.urlshorter.dao.entities.ShortenedUrl;
+
+import javax.validation.constraints.NotBlank;
+
+public record UrlGenerateRequestDto(@NotBlank String url, @NotBlank String title) {
```

Comment:
just `UrlInfoDto` or `ShortUrlRequestDto` ... `generate` seems extra

Reply:
This dto naming with the operation I've taken from our prod to define different operations with different required optional/required attributes. I agree that here it seems redundantly, will change it

Reply:
Also, because we have different dtos with different attributes we have too many converters for these DTOs. 
What do you this, it there any way to unite this logic? For me, the only way to optimize it - is to use GraphQL. 
Do you know any other alternatives to optimize it?

Reply:
upd coomit: added mapstruct